backend/src/directors/report_director.py

```diff
@@ -1,27 +1,37 @@
+import logging
+import uuid
 from fastapi import UploadFile
 
+from src.llm.llm import LLMFile
 from src.session.file_uploads import FileUploadReport, store_report
-from src.utils.file_utils import handle_file_upload
 from src.agents import get_report_agent, get_materiality_agent
 
+logger = logging.getLogger(__name__)
 
 async def report_on_file_upload(upload: UploadFile) -> FileUploadReport:
 
-    file = handle_file_upload(upload)
+    file_stream = await upload.read()
+    if upload.filename is None:
+        raise ValueError("Filename cannot be None")
```

Comment:
@IMladjenovic @mic-smith would this exception be better as a HTTPException so that the error goes straight back to the ui?